src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -554,6 +561,78 @@ public CodeNamespace CreateSourceModel(OpenApiUrlTreeNode? root)
         return rootNamespace;
     }
 
+    private void AddOperationSecurityRequirementToDOM(OpenApiOperation operation, CodeClass codeClass)
+    {
+        if (openApiDocument is null)
+        {
+            logger.LogWarning("OpenAPI document is null");
+            return;
+        }
+
+        if (operation.Security == null || !operation.Security.Any())
+            return;
+
+        var securitySchemes = openApiDocument.Components.SecuritySchemes;
+        foreach (var securityRequirement in operation.Security)
+        {
+            foreach (var scheme in securityRequirement.Keys)
+            {
+                var securityScheme = securitySchemes[scheme.Reference.Id];
+                switch (securityScheme.Type)
+                {
+                    case SecuritySchemeType.Http:
+                        AddHttpSecurity(codeClass, securityScheme);
+                        break;
+                    case SecuritySchemeType.ApiKey:
+                        AddApiKeySecurity(codeClass, securityScheme);
+                        break;
+                    case SecuritySchemeType.OAuth2:
+                        AddOAuth2Security(codeClass, securityScheme);
+                        break;
+                    default:
+                        logger.LogWarning("Unsupported security scheme type: {Type}", securityScheme.Type);
+                        break;
+                }
+            }
+        }
+    }
+
+    private void AddHttpSecurity(CodeClass codeClass, OpenApiSecurityScheme securityScheme)
```

Comment:
opportunity to refactor to more concise code with the switch since all that's changing is the name